src/api.v2/helpers/worker/workSubmit/submitMarkerHeatmapWork.js

```diff
@@ -1,9 +1,16 @@
-const getExtraDependencies = require('./getExtraDependencies');
-const submitWork = require('./submitWork');
+const getCellSetsAffectingDownsampling = require('./getCellSetsAffectingDownsampling');
+const submitWorkForHook = require('./submitWorkForHook');
+const bucketNames = require('../../../../config/bucketNames');
+const getObject = require('../../s3/getObject');
 
 const submitMarkerHeatmapWork = async (message) => {
   const { experimentId, input: { authJWT } } = message;
 
+  const { cellSets } = JSON.parse(await getObject({
+    Bucket: bucketNames.CELL_SETS,
+    Key: experimentId,
+  }));
+
```

Comment:
Do we need to get the cellsets and overwrite the value of `body.downsampleSetttings.cellSets` with them because `body.downsampleSettings.cellSets` are out of date? If so, why are they out of date? Why aren't we fetching the correct cell sets from the worker (I believe we were or still are fetching cell sets from the worker to aid computation of tasks).

Reply:
thanks for the explanation, makes sense